pyanaconda/task/task.py

```diff
@@ -0,0 +1,203 @@
+# DBus Task interface.
+#
+# Base class of tasks.
+# Task is used by modules to implement asynchronous time consuming installation
+# or configuration tasks.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from threading import Lock
+from abc import ABC, abstractmethod
+from pyanaconda.constants import THREAD_DBUS_TASK
+from pyanaconda.task.task_interface import TaskAlreadyRunningException
+from pyanaconda.threading import threadMgr, AnacondaThread
+from pyanaconda.async_utils import async_action_nowait
+
+__all__ = ['Task', 'TaskAlreadyRunningException']
+
+
+class Task(ABC):
+    """Base class implementing DBus Task interface."""
+
+    _task_counter = 1
+
+    def __init__(self, dbus_modul_path):
+        super().__init__()
+        self._progress = (0, "")
+
+        self.__cancel_lock = Lock()
+        self.__cancel = False
+
+        self._task_number = Task._task_counter
+        Task._task_counter += 1
+
+        self._running_changed_signal = None
+        self._progress_changed_signal = None
+        self._error_raised_signal = None
+
+        self._dbus_name = "{}.{}".format(dbus_modul_path, self._task_number)
```

Comment:
I think that we should find a better solution for publishing objects and creating and keeping DBus names. But that is for another issue.

Reply:
Could be nice topic for a future discussion.